lib/puppet/gettext/config.rb

```diff
@@ -126,7 +137,6 @@ def self.load_translations(project_name, locale_dir, file_format)
   def self.add_repository_to_domain(project_name, locale_dir, file_format)
     # check if we've already loaded these transltaions
     current_chain = FastGettext.translation_repositories[FastGettext.text_domain].chain
-    return current_chain if @loaded_repositories[project_name]
```

Comment:
Was this intended? The comment may not match anymore

Reply:
I saw a later commit removed the comment. But it seems `add_repository_to_domain` will always append the repository to the chain. Will that lead to the same repo being added multiple times?

Reply:
Yes, sometimes, but that's intentional. In the command line case, we want to load translations both from the modulepath and from vardir, since as far as I can tell we check both places for module additions (like custom facts). If a module with the same name is present in both places, we want to check both for translations. This is why we are pretty aggressive about clearing out text domains each time an environment is initialized.

Reply:
Also about returning nil, in theory it can, if both the requested text domain and the default text domain don't exist, but that will mean we've set up something wrong and shouldn't be able to continue.